samples/tests/ComponentExamples/ComponentExamples.test.ts

```diff
@@ -0,0 +1,56 @@
+import { test, expect } from '@playwright/test';
+import { encodeQueryData, createChatThreadAndUsers } from '../common/utils';
+import * as path from 'path';
+import dotenv from 'dotenv';
+
+dotenv.config({ path: path.join(__dirname, '..', '.env') });
+
+const SERVER_URL = 'http://localhost:8080';
+const PARTICIPANTS = ['Jack'];
+
+const dataUiId = (id: string): string => `[data-ui-id="${id}"]`;
+
+test.describe('Component Examples Test', () => {
+  test.beforeEach(async ({ page }) => {
+    // Go to the starting url before each test.
+    const user = await createChatThreadAndUsers(PARTICIPANTS);
+
+    const qs = encodeQueryData(user[0]);
+    const url = `${SERVER_URL}?${qs}`;
+    await page.goto(url);
+  });
+
+  // only need to make sure it is loading correctly, the rest will be the responsibility of composite tests
+  test('Whether html page is loaded right with 2 composites', async ({ page }) => {
+    expect(await page.waitForSelector('text=Connect components to StatefulClient')).toBeTruthy();
+    expect(await page.waitForSelector('text=Render single video stream in your own component')).toBeTruthy();
+    // Wait for message to finish sending
+    await page.waitForTimeout(1000);
+    await page.type(dataUiId('sendbox-textfield'), 'Hello!');
+    await page.keyboard.press('Enter');
+
+    await page.waitForSelector(`[data-ui-status="delivered"]`);
+
+    await page.addScriptTag({
+      content: `document.querySelector('[data-ui-id=message-timestamp]').innerText='timestamp';`
+    });
+
+    // Start local video using custom start button
+    await page.click(dataUiId('custom-start-button'));
+    await page.waitForFunction(() => {
+      const videoNode = document.querySelector('video');
+      return document.querySelectorAll('video').length === 1 && videoNode?.readyState === 4;
```

Comment:
Why `4`? Can you use an enum value here? If not, please document which enum, what value.

Reply:
change it to HTMLMediaElement.HAVE_ENOUGH_DATA according to api doc from https://developer.mozilla.org/en-US/docs/Web/API/HTMLMediaElement/readyState